wordpress-mobile/WordPress-iOS
# Media Editor Architecture Improvement

This PR improves the Media Editor architecture to:

* Make easier to add new capabilities in the future
* Make easier to deal with async images (next task)
* Make the code better and decoupled :)

## A little bit of information

The Media Editor is now modular, which means new capabilities can be added in the form of plugins! Yay!

Here's an image that shows that concept:

![Artboard](https://user-images.githubusercontent.com/7040243/70556555-e5d4e800-1b5f-11ea-8537-ed6d46836519.png)

This makes really easy for us or the community to grow the list of features of our Media Editor.

## To test

Nothing should have changed from this PR:

1. Using Aztec, add an image
2. Crop the image, check if the image was updated and the log of events is ok

PR submission checklist:
